presto-main/src/main/java/com/facebook/presto/operator/TableScanOperator.java

```diff
@@ -205,6 +209,9 @@ public boolean isFinished()
         if (!finished) {
             createSourceIfNecessary();
             finished = (source != null) && source.isFinished();
+            if (source != null) {
+                systemMemoryContext.setBytes(source.getSystemMemoryUsage());
```

Comment:
This need to exist because `source` may have been created 2 lines up. Also, `isFinished` could potentially change system memory usage.